test/then.js

```diff
@@ -0,0 +1,44 @@
+var assert = require('assert');
+var Q = require('q');
+
+var R = require('..');
+var eq = require('./shared/eq');
+
+
+describe('then', function() {
+  it('invokes then on the promise with the function passed to it', function(done) {
+    var p = Q(1);
+
+    var f = function(n) {
+      eq(n, 1);
+      done();
+    };
+
+    R.then(f, p)
+  });
+
+  it('throws a typeError if the then method does not exist', function() {
+    assert.throws(
+      function() { R.then(R.inc, 1); },
+      function(err) {
+        return err.constructor === TypeError &&
+               err.message === '1 does not have a method named "then"';
+      }
+    )
+
+  });
+
+  it('works with any object that has a then method', function() {
+    var thennable = {
+      then: function(f) {
+        return f(42);
+      }
+    };
+
+    var f = function(n) {
+      eq(n, 42);
```

Comment:
I think we need `done()` here. Otherwise, the test would pass if `f` is never called.